### ISSUE TITLE
Error loading Google MAP in Web

### ISSUE THREAD

**Issue by harsuu942** [a14n/dart-google-maps#104]
TypeError: Cannot read property 'maps' of undefined.
How can I solve this?

**Comment by a14n** [comment]
Did you correctly include the Maps API JavaScript using a script tag before your dart script ?

```html
<script src="https://maps.googleapis.com/maps/api/js?key=API_KEY"></script>
```

**Comment by harsuu942** [comment]
Yes, I have already included the API javaScript but still showing TypeError.

**Comment by a14n** [comment]
It looks like an issue with the html setup. Do you have something similar to https://github.com/a14n/dart-google-maps/blob/master/example/01-basics/map-simple/map.html ?

Could you post your html code?

**Comment by harsuu942** [comment]
Here is my HTML code.


<!DOCTYPE html>
<html>
<head>
  <base href="/">

  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
  <meta name="description" content="A new Flutter project.">

  
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  <meta name="apple-mobile-web-app-title" content="web">
  <link rel="apple-touch-icon" href="icons/Icon-192.png">

  <title>web</title>
  <link rel="manifest" href="manifest.json">

  <style>
      html, body, #map {
        height: 100%;
        margin: 0px;
        padding: 0px
      }
  </style>

</head>
<body>

<div id="map"></div>
<script src="https://maps.googleapis.com/maps/api/js?key=googleAPIKey"></script>
 <script defer src="page.dart.js"></script>

</body>
</html>

**Comment by a14n** [comment]
Hard to tell what is wrong without additional informations. If you have a like to a public repo I could access I can take a look but for now I can't help more.

**Comment by harsuu942** [comment]
The sample example is working fine but I want to use it as a widget.

Here is what I used as a widget

Widget getMap() {
  String htmlId = "map";

  // ignore: undefined_prefixed_name
  ui.platformViewRegistry.registerViewFactory(htmlId, (int viewId) {
    final myLatlng = LatLng(1.3521, 103.8198);

    final mapOptions = MapOptions()
      ..zoom = 10
      ..center = LatLng(1.3521, 103.8198);

    final elem = DivElement()
      ..id = htmlId
      ..style.width = "100%"
      ..style.height = "100%"
      ..style.border = 'none';



    final map = GMap(elem, mapOptions);

    Marker(MarkerOptions()
      ..position = myLatlng
      ..map = map
      ..title = 'Hello World!'
    );

    return elem;
  });

  return HtmlElementView(viewType: htmlId);
}

 This time getting error: "TypeError: Cannot read property 'maps' of undefined" so can't use as a widget
 so could please help me out.

**Comment by a14n** [comment]
Hard to tell what is wrong. You can take a look at how [google_maps_flutter_web works](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart).

**Comment by harsuu942** [comment]
Thanks a man !!! I got the solutions.
